rasa/utils/train_utils.py

```diff
@@ -534,6 +705,71 @@ def sample_negatives(
     )
 
 
+def tf_matmul_sparse(inputs: tf.SparseTensor, kernel: tf.Tensor):
```

Comment:
could you please add comments and a link where we get it from

Reply:
I took a closer look, it is for 3Dx3D matmul. I think it is simpler if we do the same trick with flattening we do in negative sample, since we need 3Dx2D